src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -20,21 +20,24 @@ namespace Microsoft.CodeAnalysis.PublicApiAnalyzers
 {
     public partial class DeclarePublicApiAnalyzer : DiagnosticAnalyzer
     {
-        private sealed class ApiLine
+        private sealed record AdditionalFileInfo(string Path, SourceText SourceText, bool IsShippedApi);
+
+        private readonly record struct ApiLine
```

Comment:
made into a simple struct, which also has a pointer to the common data stored across all lines from teh same file.